urwid/widget/scrollable.py

```diff
@@ -481,7 +481,14 @@ def __init__(
         """
         if Sizing.BOX not in widget.sizing():
             raise ValueError(f"Not a box widget: {widget!r}")
-        if not isinstance(widget, SupportsScroll):
+
+        def orig_iter(w: Widget) -> Iterator[Widget]:
+            yield w
+            while hasattr(w, "original_widget"):
```

Comment:
`w.original_widget` can point to `w`, we need to filter such behaviour to prevent infinite loop